homebridge/thermostat.ts

```diff
@@ -0,0 +1,237 @@
+import { Logging, PlatformAccessory } from 'homebridge'
+import { Observable } from 'rxjs'
+import { distinctUntilChanged, map, switchMap } from 'rxjs/operators'
+
+import { RingDevice } from '../api'
+import { RingDeviceType, ThermostatMode } from '../api/ring-types'
+import { BaseDeviceAccessory } from './base-device-accessory'
+import { RingPlatformConfig } from './config'
+import { hap } from './hap'
+
+export class Thermostat extends BaseDeviceAccessory {
+  private onTemperature: Observable<number | undefined>
+
+  constructor(
+    public readonly device: RingDevice,
+    public readonly accessory: PlatformAccessory,
+    public readonly logger: Logging,
+    public readonly config: RingPlatformConfig
+  ) {
+    super()
+
+    const { Characteristic, Service } = hap
+
+    // Component Device (Temperature Sensor)
+
+    this.onTemperature = this.device.onComponentDevices.pipe(
+      switchMap((devices) => {
+        const temperatureSensor = devices.find(
+          ({ deviceType }) => deviceType === RingDeviceType.TemperatureSensor
+        )
+        if (!temperatureSensor) {
+          return []
+        }
+        this.logger.debug(
+          `Discovered a component temperature sensor for ${this.device.name}`
+        )
+        return temperatureSensor.onData.pipe(
+          map(({ celsius: temperature }) => {
+            this.logger.debug(
+              `Component temperature sensor for ${this.device.name} reported ${temperature} degrees`
+            )
+            return temperature
+          }),
+          distinctUntilChanged()
+        )
+      })
+    )
+
+    // Required Characteristics
+
+    this.registerObservableCharacteristic({
+      characteristicType: Characteristic.CurrentHeatingCoolingState,
+      serviceType: Service.Thermostat,
+      onValue: this.onTemperature.pipe(
+        map((temperature) => {
+          const { setPoint, mode } = this.device.data
```

Comment:
Spotted a regression—`CurrentHeatingCoolingState` is re-evaluated when `this.device.data.setPoint` or `this.device.data.mode` change.

Reply:
Resolved by commits https://github.com/dgreif/ring/pull/331/commits/bad31ecc23e34f9d44b3f1bef4983d79c9828961 and https://github.com/dgreif/ring/pull/331/commits/bca661528787802e0174fa1f37af7fb39ce92079

Reply:
Verified this works as expected. 👍